kpops/utils/pydantic.py

```diff
@@ -111,20 +113,48 @@ def json_schema_extra(schema: dict[str, Any], model: type[BaseModel]) -> None:
 class YamlConfigSettingsSource(PydanticBaseSettingsSource):
     """Loads variables from a YAML file at the project's root."""
 
-    path_to_config = Path("config.yaml")
+    log = logging.getLogger()
+
+    path_to_settings = Path()
+    settings_file_base_name = "config"
+    environment: str | None = None
+
+    def __init__(self, settings_cls):
+        super().__init__(settings_cls)
+        # Check if path to settings yaml definition is valid
+        if not self.path_to_settings.exists():
+            msg = f"Path to config directory {self.path_to_settings} does not exist."
+            raise ValueError(msg)
+        elif self.path_to_settings.is_file():
+            msg = f"Path to config directory {self.path_to_settings} must point to a directory."
+            raise ValueError(msg)
+        default_settings = self.__load_settings(
+            self.path_to_settings / f"{self.settings_file_base_name}.yaml"
+        )
```

Comment:
Extract to method and call it `check_if_path_config_yaml_is_valid`

Reply:
Only extracted the 2 `if`s and chose a different name. Do you agree with the change?